lib/awful/tag.lua

```diff
@@ -1350,11 +1352,35 @@ capi.tag.add_signal("property::urgent")
 
 capi.tag.add_signal("property::urgent_count")
 capi.tag.add_signal("property::volatile")
+capi.tag.add_signal("removal-pending")
 
 capi.screen.add_signal("tag::history::update")
 
 capi.screen.connect_signal("tag::history::update", tag.history.update)
 
+capi.screen.connect_signal("removed", function(s)
+    -- First give other code a chance to clean up
+    for _, t in pairs(s.tags) do
+        t:emit_signal("removal-pending")
```

Comment:
I know I could implement `request::tag` myself using this signal, but I would prefer it to be integrated into the "default" sequence.

First, tags should emit a `request::screen` without any handler by default. If the screen changed, then return. After that, the client should each emit `request::tag`. The default handler will already do less or more what's your doing here. This way, the tag can be salvaged with its layout and all and just moved to another screen.